protoc-gen-grpc-gateway/gengateway/template.go

```diff
@@ -423,6 +424,7 @@ func Register{{$svc.GetName}}{{$.RegisterFuncSuffix}}Client(ctx context.Context,
 				}
 			}(ctx.Done(), cn.CloseNotify())
 		}
+	{{- end }}
```

Comment:
I'm not opposed to stripping this entirely, but if we go that route, it seems like we should also completely remove support for Go < 1.7 by removing the UseRequestContext and the CLI flag so we don't end up with any inadvertent leaks/regressions.

I'd elected to go this way at first to minimize the risk of silent regressions with older Go releases, since the existence of this flag seems to imply some degree support for older Go releases.

Thoughts?

Reply:
To me the choice of using the request context is still relevant today, since you may want to decouple the cancellation from that of the incoming HTTP requests.

Reply:
I see, and entirely fair! I'd not considered that view point. Adjusting.